ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/machine/WsAgentStateController.java

```diff
@@ -86,16 +100,34 @@ public void initialize(DevMachine devMachine) {
 
     @Override
     public void onClose(WebSocketClosedEvent event) {
-        Log.info(getClass(), "Test WS connection closed with code " + event.getCode() + " reason: " + event.getReason());
         if (state.equals(STARTED)) {
-            state = STOPPED;
-            eventBus.fireEvent(WsAgentStateEvent.createWsAgentStoppedEvent());
+            checkWsAgentState();
         }
     }
 
+    private void checkWsAgentState() {
+        final String url = restContext + "/workspace/" + devMachine.getWorkspace() + "/check";
```

Comment:
Can not we put and take this URL from DevMachine?
Such a string concatenation looks bad.
BTW, DevMachine is initialized not in constructor, what if DevMachine == null by any reason?